### PR TITLE
xero-updates-july

### PR DESCRIPTION
**Pull request by jasper-bartlett**
# Description

* Creation of several new pages relating to Xero certification
* edits to some existing pages on Xero certification

## Type of change

- [ ] New document(s)/updating existing

## Reviews and merging

You are responsible for getting your PR merged. Address review comments promptly and **make sure to merge the PR when ready**.
Feel free to 'Enable automerge' - your PR will automatically merge when accepted and passing the build.

### CODE REVIEW

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+
+**Action Required: Significant**
+
+If an error occurs, your UI should communicate that fact to the user. It is best practice to provide detailed error messages or logs. Codat provides detailed error messaging for any issues with the integration, including status codes, listed [here](/using-the-api/errors).
+
+### 5. Offline access
+
+**Action Required: None**
+
+This checkpoint is fully handled by Codat as we handle the connectivity with Xero's API.
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:56]: For every occurrence of this sentence, would it be possible to give some background on what the checkpoint actually wants, or link out to Xero documents? 

I also suggest rewording it to "We handle this checkpoint in full in our Xero integration".

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+If you plan to create, update or delete data in Xero, you need to create an account mapping interface for your users. In preparation, we recommend you review Xero's [best practice guide](https://developer.xero.com/documentation/guides/how-to-guides/integration-best-practices) and Codat's [implementation guide](/using-the-api/best-practices/implementing-a-mapping-page).
+
+### 13. Currency
+
+**Action Required: None or Minimal**
+
+This checkpoint is only relevant if your app deals with multiple currencies. 
+
+Our currency data type is available across all integrations and is aware of the currencies enabled by the SME in Xero. When creating or updating transactional data, such as a payment in another currency, use our [Get push options](/codat-api#/operations/get-create-update-model-options-by-data-type) endpoint to see which currencies have been set up by your customer in Xero.
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:114]: I am not sure what the first sentence here is supposed to mean. Aware of the currencies?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+
+---
+
+## Preferred user experience
+
+The following checkpoints are considered best practices, and not requirements. These are not mandatory for the Xero certification, but can enhance the user experience. 
+
+### 17. Sign in with Xero
+
+If you would like to implement Sign in with Xero using Codat, please contact your Codat account team for detailed guidance.
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:154]: Is there a link to Xero that we can give to explain what Sign in with Xero is?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+
+### 18. Deep linking
+
+Deep linking enables your application to create a link that directs the user to a specific resource within a specific Xero organization. This provides a quick and easy way for the user to jump into Xero from the resource referenced in your product.
+
+You can view the direct links in the `sourceUrls` property of our [Get company info](/accounting-api#/operations/get-company-info) endpoint. 
+
+We also use Xero record ids, which means you can implement deep links for Xero contacts by using a hyperlink with the following structure:
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:162]: what is Xero record ids? Is this a property of some sort? is it on our side, or on Xero's side? should it be formatted as an endpoint property name, like `recordId`?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+
+`https://go.xero.com/Contacts/View/{Codat's customerId}`
+
+You can find further information about deep linking in Xero's [own documentation](https://developer.xero.com/documentation/guides/how-to-guides/deep-link-xero/).
+
+### 19. Invoice status
+
+Available when pushing via the Invoice endpoint
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:170]: Not enough background here for me. What are we trying to say here?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+### 3. Connection
+
+**Action required: significant**
+
+While Codat handles the initial connection to Xero via the Codat Link UI, Xero requires that you enable your user to manage this connection on an ongoing basis. For example, you can create a page in your app to satisfy these requirements: 
+
+| Requirement | Recommendations |
+| :-- | :-- |
+| Display the name of the tenant that has been connected | This can be retrieved from our [Get company info](/accounting-api#/operations/get-company-info) endpoint. |
+| Display the current status of the connection(s). If disconnected, provide a button to reconnect to Xero | Status - `Connection.status` endpoint.  <br/>Link to power your button to connect - `Connection.linkURL` |
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:40]: For the connection.status and connection.linkURL, are these codat or xero endpoints?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+This checkpoint assumes that your borrowers are able to manage their data connection via a digital portal you provide. This may not be relevant if:
+  * You pull data once during customer onboarding and then disconnect.
+  * You do not have a digital lending journey for borrowers and rely on a different approach instead (for example, emails).
+
+  If either of these are true, communicate this clearly to Xero during your partner application. Otherwise, your digital portal UI should satisfy the following requirements:
+
+| Requirement | Recommendations |
+| :-- | :-- |
+| Display the name of the tenant that has been connected | This can be retrieved from our [Get company info](/accounting-api#/operations/get-company-info) endpoint. |
+| Display the current status of the connection(s). If disconnected, provide a button to reconnect to Xero | Status - `Connection.status` endpoint.  <br/>Link to power your button to connect - `Connection.linkURL` |
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md:53]: Here is the same question on the connection.status and connection.linkURL - and generally questions that apply to the App Partner page also apply to this page

File: docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+### 20. Invoice URL and bank transactions URL
+
+This is not currently supported via Codat. Please contact your Codat account team with a feature request if this functionality is required for your use case. 
+
+### 21. Logs
+
+You can [retrieve a log](/using-the-api/pull-history) of all data types and past syncronizations performed by Codat, and display it back to the user, using our [List pull operations](/codat-api#/operations/list-pull-operations) endpoint. This may be useful if, for example, they want to know when invoices were last fetched from their Xero instance.
+
+You can also [retrieve and display the history](/configure/portal/pull-and-push-history) of created, updated or deleted data using our [List push operations](/codat-api#/operations/get-company-push-history) endpoint.
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md:184]: A general suggestion is having a Read next section at the bottom with the links to all 4 pages that constitute the Partner certification scope of materials

File: docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
+---
+title: "Sign Up with Xero"
+description: "Learn how to enable the Sign Up with Xero flow with Codat to support your app's certification"
+---
+
+Providing Sign Up with Xero (SUwX) is a requirement for all App Store Partners seeking certification, as prescribed by Xero’s [certification checkpoints](https://developer.xero.com/documentation/xero-app-store/app-partner-guides/certification-checkpoints/#required-for-all-integrations). It is optional for Financial Services Partners, depending on whether you choose to be listed in the App Store.
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md:6]: Are we keen to use this acronym? Looks quite intimidating and out of place in headers. We don't use the name of the functionality that often, so perhaps okay to just refer to it as Sign Up with Xero flow every time. Do Xero themselves shorten it to anything?

File: docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+* Option A  
+
+  This option uses a related concept called [Sign In with Xero](https://developer.xero.com/documentation/xero-app-store/app-partner-guides/sign-in,) which enables users to use Xero as an identity provider when logging into your app. This is similar to signing in with Google or Facebook. 
+  
+  This approach typically requires significantly more developer resource and is not possible for all apps (for example, apps that do not have a completely self-serve onboarding process and need customers to speak to a sales representative first).
+ 
+* Option B
+  
+  This approach is more popular with Codat clients. It is simpler and enables apps to reduce the time and developer resource required to pass Xero Certification.
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md:22]: With option A, we give some info on what the technical detail of it is - like using the related sign in with xero concept. is there a one-sentence description we could give to option B?

File: docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+description: "Learn how to enable the Sign Up with Xero flow with Codat to support your app's certification"
+---
+
+Providing Sign Up with Xero (SUwX) is a requirement for all App Store Partners seeking certification, as prescribed by Xero’s [certification checkpoints](https://developer.xero.com/documentation/xero-app-store/app-partner-guides/certification-checkpoints/#required-for-all-integrations). It is optional for Financial Services Partners, depending on whether you choose to be listed in the App Store. 
+
+SUwX is a user journey that takes potential customers from the Xero App Store directly to your app. It reduces the number of steps users need to take to sign up to your app and enables Xero to track referrals that originate in the Xero App Store.
+
+## SUwX implementation options
+
+Xero’s documentation outlines [two approaches to Sign Up with Xero](https://developer.xero.com/documentation/xero-app-store/app-partner-guides/sign-up):
```
Comment (pzaichkina) [inline docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md:12]: Xero also calls them Modified and Recommended flow - perhaps we should stick to those names?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md (hunk 6 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:40]: I don't actually know. This isn't something I wrote. It was created by Solutions in the previous version. I'll check with them to verify first.

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md (hunk 1 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:56]: My instinct was to keep this page as lean as possible. I want to give people the simplest, shortest guidance and avoid repeating the content in the Xero docs which lays out the requirements in full. 

This is explicitly because the Xero document is long and I have observed clients not reading it in full and missing things. 

This page is intended as a companion guide + simple checklist so they know what to do. It will be shared with clients in that context. If you feel that is insufficiently clear to someone who finds the page without that context, perhaps we could add a contextualising sentence at the top? Something like "Please Xero's Certification Checkpoints in full to understand Xero's requirements. This guide is intended as a companion piece, to help you understand what actions you need to take."

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md (hunk 2 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:114]: Some apps deal with multiple currencies. 

If the reader's app is not multi-currency, they can ignore this requirements. 
If the reader's app is multi-currency, they need to ensure they follow the guidance on line 114. 

How do you think that can be better expressed?

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md (hunk 5 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:170]: Tbh none of this info on the '## Preferred user experience section is written by me. It is all brought over from the old docs. 

I would be happy to delete all of this and simple end with:

## Preferred user experience

The following checkpoints are considered best practices. These are not mandatory for the Xero certification, but can enhance the user experience. 

If you would like guidance on how to comply with any of these checkpoints, please consult your Codat account team.

File: docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md [hunk 12]
```diff
@@ -0,0 +1,184 @@
+---
+title: "Xero's certification checkpoints"
+description: "A guide to getting Xero's Lending Partnership certification with Codat"
+---
+
+For lenders that want to become Xero Partners, Xero stipulates two technical requirements to comply with:
```
[the remark this answers is not in the document]
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md:6]: Xero is not consistent in their capitalisation. 

Nor does Xero does not use the phrase 'Lending Partner' in their public docs. They never get more specific than 'Financial Services Partner' or 'App Store Partner' in their documentation. But they do mention 'Lending Partner' in other PDF guidance that they give specifically to lenders. 

I recommend keeping Lending Partner as it is but it's not a strong opinion if you prefer to match an in house Codat style.

File: docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md (hunk 8 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-lending.md:185]: Ok thanks - I'll have a think about that, once we've created a V1 of all the docs we need.

File: docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md (hunk 9 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md:6]: I'm happy to use the full name if you are!

File: docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md (hunk 11 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md:12]: No - this one is important! Codat currently doesn't enable the Recommended flow. 

The language here is balanced to not make that overt (we can build it if a client needs it, or if Xero asks) but we want to do everything we can to reassure clients that the simpler approach is ok (which it really is. everyone does it. Xero just uses intense language to push people towards 'recommended')

File: docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md (hunk 6 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/checkpoints-app-store.md:40]: https://api.codat.io/swagger/index.html#/Connection/get_companies__companyId__connections__connectionId_

File: docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md (hunk 10 above)
Comment (jasper-bartlett) [reply docs/integrations/accounting/xero/partner-certification/sign-up-with-xero.md:22]: Given we use the rest of the document to explain Option B, is that necessary? 

Currently, clients get confused by the fact that we don't mention Option A anywhere in our docs and only present a solution for Option B. 

I want to use this page to avoid clients being confused about which option to do (which they are atm) and then guide them through option B (which all of the last 25 clients have done. no one has ever wanted to do Option A with Codat)

**Review by pzaichkina — APPROVED** [review]
